### PR TITLE
Load remote and local images for FOMOD

### LINKED ISSUE #614: Rework `AssetUrl` into a discriminated union for different types of assets
Right now, the `AssetUrl` is just a string underneath, but it can be a relative path or a URL to some remote image. A discriminated union would be ideal, as it allows you to have one type that is a union of other types, which you can check at runtime.

This can be done using [`OneOf`](https://github.com/mcintyre321/OneOf) or similar libraries.

The new type should also make loading the data easier. Either integrate with the existing `IStreamFactory` interface or just add a method that returns `byte[]`, so the FOMOD UI can easily create a `Bitmap` out of the data.

### PR DESCRIPTION
**Pull request by erri120**
Fixes #614.

### CODE REVIEW

File: src/Games/NexusMods.Games.FOMOD/CoreDelegates/InstallerDelegates.cs [hunk 1]
```diff
@@ -12,12 +13,19 @@ public sealed class InstallerDelegates : ICoreDelegates
     public IContextDelegates context { get; }
     public IIniDelegates ini => throw new NotImplementedException();
     public IPluginDelegates plugin { get; }
-    public IUIDelegates ui { get; }
 
-    public InstallerDelegates(ILoggerFactory loggerFactory, IGuidedInstaller guidedInstaller)
+    public IUIDelegates ui => UiDelegates;
+    public readonly UiDelegates UiDelegates;
```
Comment (Al12rs) [inline src/Games/NexusMods.Games.FOMOD/CoreDelegates/InstallerDelegates.cs:18]: Why was this needed?

**Comment by Al12rs** [comment]
Apart from the question above, everything else looked good to me, but some parts will require rewriting when #615 is merged.

File: src/Games/NexusMods.Games.FOMOD/CoreDelegates/InstallerDelegates.cs (hunk 1 above)
Comment (erri120) [reply src/Games/NexusMods.Games.FOMOD/CoreDelegates/InstallerDelegates.cs:18]: The FOMOD installer needs to set a field inside the `UiDelegates` class.

**Review by Al12rs — APPROVED** [review]
